libgnucash/engine/gnc-budget.cpp

```diff
@@ -631,25 +700,36 @@ gnc_budget_set_account_period_note(GncBudget *budget, const Account *account,
 
 }
 
-gchar *
-gnc_budget_get_account_period_note(const GncBudget *budget,
-                                   const Account *account, guint period_num)
+static std::string
+get_account_period_note (const GncBudget *budget,
+                         const Account *account, guint period_num)
 {
     gchar path_part_one [GUID_ENCODING_LENGTH + 1];
     gchar path_part_two [GNC_BUDGET_MAX_NUM_PERIODS_DIGITS];
     GValue v = G_VALUE_INIT;
-    gchar *retval;
+    std::string retval;
 
-    g_return_val_if_fail(GNC_IS_BUDGET(budget), NULL);
-    g_return_val_if_fail(account, NULL);
+    g_return_val_if_fail (GNC_IS_BUDGET(budget), "");
+    g_return_val_if_fail (account, "");
 
     make_period_path (account, period_num, path_part_one, path_part_two);
     qof_instance_get_kvp (QOF_INSTANCE (budget), &v, 3, GNC_BUDGET_NOTES_PATH, path_part_one, path_part_two);
-    retval = G_VALUE_HOLDS_STRING (&v) ? g_value_dup_string(&v) : NULL;
+    if (G_VALUE_HOLDS_STRING (&v))
+        retval = g_value_get_string(&v);
     g_value_unset (&v);
     return retval;
 }
 
+
+gchar *
+gnc_budget_get_account_period_note (const GncBudget *budget,
+                                    const Account *account, guint period_num)
+{
+    g_return_val_if_fail (period_num < GET_PRIVATE(budget)->num_periods, nullptr);
+    auto& data = get_perioddata (budget, account, period_num);
+    return data.note.empty () ? nullptr : g_strdup (data.note.c_str());
```

Comment:
Should this function return `nullptr` if `!data.value_is_set`? IOW does it make sense for a period to have a valid note but not a valid value?

Reply:
Dunno... I've replicated existing behaviour...

Reply:
OK, I guess, since none of us actually uses budgets. :-/